spark3/src/main/java/org/apache/iceberg/spark/source/IcebergSource.java

```diff
@@ -62,7 +62,7 @@ public SparkTable getTable(StructType schema, Transform[] partitioning, Map<Stri
     Table icebergTable = getTableAndResolveHadoopConfiguration(options, conf);
 
     // Build Spark table based on Iceberg table, and return it
-    return new SparkTable(icebergTable, schema);
+    return new SparkTable(icebergTable, schema, options);
```

Comment:
I think that `SparkTable` should include a `snapshotId` so that it returns the correct snapshot's schema to the analyzer, when possible. I don't think that it needs to include these options, when no other options are used by the table.

Reply:
Ok, I can do that.

Reply:
I extracted `snapshotId` and `asOfTimestamp` from `options` here if they appear and pass the values to `SparkTable` in its constructor instead of having `SparkTable` do that. I honestly think it's just half a dozen of one and six of the other.